Objects/funcobject.c

```diff
@@ -665,6 +665,20 @@ func_get_type_params(PyFunctionObject *op, void *Py_UNUSED(ignored))
     return Py_NewRef(op->func_typeparams);
 }
 
+static int
+func_set_type_params(PyFunctionObject *op, PyObject *value, void *Py_UNUSED(ignored))
+{
+    /* Not legal to del f.__type_params__ or to set it to anything
+     * other than a tuple object. */
+    if (value == NULL || !PyTuple_Check(value)) {
```

Comment:
Does it make sense to use `PyTuple_CheckExact` instead of `PyTuple_Check`? With `PyTuple_Check` we can pass to `__type_params__` subclass of tuple. May it break something?

Reply:
We don't really use the `__type_params__` in the interpreter, so there's low risk. The other attributes also allow subclasses (e.g. check `func_set_defaults` and `func_set_name`).